pkg/cmd/rhoam/generate/generate.go

```diff
@@ -0,0 +1,120 @@
+package generate
+
+import (
+	"fmt"
+	v1beta1 "github.com/redhat-developer/app-services-cli/apis/3scale/v1beta1"
+
+	kafkaFlagutil "github.com/redhat-developer/app-services-cli/pkg/cmd/kafka/flagutil"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+)
+
+type options struct {
+	name       string
+	url        string
+	systemName string
+	// apicast environment {staging|production}
+	env string
+	f   *factory.Factory
+}
+
+func NewGenerateCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "generate",
+		Short:   f.Localizer.MustLocalize("rhoam.generate.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("rhoam.generate.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("rhoam.generate.cmd.example"),
+		Args:    cobra.MinimumNArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			fmt.Fprintf(os.Stdout, "Check the test directory for the generated files")
+			err := generate(opts)
+			return err
+		},
+	}
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+
+	flags.StringVar(&opts.name, "name", "", f.Localizer.MustLocalize("rhoam.generate.flag.name.description"))
+	flags.StringVar(&opts.url, "url", "", f.Localizer.MustLocalize("rhoam.generate.flag.url.description"))
+	flags.StringVar(&opts.systemName, "systemName", "", f.Localizer.MustLocalize("rhoam.generate.flag.systemName.description"))
+	flags.StringVar(&opts.systemName, "env", "", f.Localizer.MustLocalize("rhoam.generate.flag.env.description"))
+
+	_ = cmd.MarkFlagRequired("name")
+	_ = cmd.MarkFlagRequired("url")
+
+	return cmd
+}
+
+func generate(opts *options) error {
+
+	backend := &v1beta1.Backend{
+		TypeMeta: v1.TypeMeta{
+			Kind:       v1beta1.BackendKind,
+			APIVersion: fmt.Sprintf("%s/%s", v1beta1.GroupVersion.Group, v1beta1.GroupVersion.Version),
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name: fmt.Sprintf("%s-backend", opts.name),
+			Annotations: map[string]string{
+				"git.ops/managed": "true",
+			},
+			Finalizers: []string{
+				fmt.Sprintf("backend.%s/finalizer", v1beta1.GroupVersion.Group),
```

Comment:
hey, I like the idea of using gitops finalizers for this. This will ensure that the object can't be removed via UI or CR on the cluster, however, (I know it's super early and this is a WIP and so on!) it's also introducing a problem that we would need to solve somewhere (either in gitops implementation or 3scale operator), here's an example:
1. Customer creates the backend via gitops; which is dropped on to the cluster with this custom finalizer
2. 3scale operator reconciles the CR created via gitops and adds its own finalizer
When it comes to deletion, we are facing 2 possible scenarios where this might be an issue:

**Scenario 1**
I know we are agreeing on the fact that once a CR is created via GitOps it should not be interacted with via UI, but let's say this actually happened:
1. Customer deletes the backend CR triggering deletion, or deletes Tenant linked to that CR triggering cascading deletion of tenants resources (including this particular backend). 

ps. On side note; I think we should make our implementation of gitops inteligent enough to be able to tell whether a backend/product or any other capability is linked to a tenant created via GitOps or not; if tenant is not created via GitOps I don't think we should be allowing for creating this particular tenant resources via GitOps

2. 3scale operator attempts to delete the resource but is unable to since there is a finalizer that it does not recognize, which in way is good, because it protects the CR from deletion via UI or CR; but also blocks the deletion

**Scenario 2**
1. Customer wants to delete the backend via GitOps
2. Without this extra finalizer, logic to delete a CR would be already be "covered" by lets say gitops operator, CR is marked for deletion and 3scale operator takes care of the rest
3. With the custom finalizer coming from GitOps created object, we would need add some custom logic to remove the finalizer as well, this could be potentially done via 3scale operator using the annotation above? Something like if marked for deletion and has the annotation, remove all the finalizers instead of just the single one? 

Again, I know it's super early and so on, more of throwing my thoughts out here rather than anything!

Reply:
I've managed to run it successfully just now and noticed that the finalizers are the default 3scale ones, which makes the comment above irrelevant!

Reply:
Potentially we don't need to add finalizers if the operator adds the finalizer itself. I just did this here for completeness of the cr. so the deletion should work in the expected way. if gitops see that the cr is deleted it marks it for deletion in the cluster which triggers the finalizer in the normal way.

Reply:
after discussion, the finalizers get added to the cr by the operator reconcile so it's not necessary to add it here. In fact adding it here might make more scope for errors so I will remove it.